lib/Map/TableStructure.js

```diff
@@ -219,14 +219,12 @@ TableStructure.fromCsv = function(csvString, result) {
     var json = csv.toArrays(csvString, {
         onParseValue: castToScalar
     });
-    // Remove extra blank lines at the end.
-    var n = 0;
+    // Remove any blank lines.
     for (var i = json.length - 1; i >= 0; i--) {
```

Comment:
Might be simpler to use `Array.filter` here? Not sure if CSVs get long enough for it to matter, but the `json.splice` call makes this run in quadratic-time.